src/parachain/rewards.ts

```diff
@@ -9,244 +10,384 @@ import {
     decodeFixedPointType,
     DefaultVaultsAPI,
     ElectrsAPI,
+    newCurrencyId,
     newMonetaryAmount,
+    newVaultId,
     tickerToCurrencyIdLiteral,
 } from "..";
-import { WrappedCurrency } from "../types";
-import { newAccountId } from "../utils";
+import {
+    CollateralCurrency,
+    CollateralIdLiteral,
+    currencyIdToMonetaryCurrency,
+    WrappedCurrency,
+    WrappedIdLiteral,
+} from "../types";
 
 export interface RewardsAPI {
     /**
      * @param vaultId The account ID of the staking pool nominee
      * @param nominatorId The account ID of the staking pool nominator
+     * @param collateralCurrencyId Collateral currency used by the vault
      * @returns A Monetary.js amount object, representing the reward in the given currency
      */
     computeRewardInStakingPool(
-        vaultId: string,
-        nominatorId: string
+        vaultAccountId: AccountId,
```

Comment:
General comment on the function interfaces here: the ordering of the arguments should be consistent. Here, the accountId, collateralCurrency, nominator et.c are switched around between different interfaces. I suggest you stick with something like:
1. VaultId
2. collateralCurrency
3. NominatorId
4. ...

Reply:
If we merge the vaultId and collateralCurrency into an `InterbtcPrimitivesVaultId`, the interface would be simpler